src/xrc/src/api/test.rs

```diff
@@ -168,6 +166,18 @@ fn btc_queried_exchange_rate_mock() -> QueriedExchangeRate {
     )
 }
 
+/// A simple mock BTC/USDT [QueriedExchangeRateWithFailedExchanges].
+fn btc_queried_exchange_rate_with_failed_exchanges_mock(exchanges: Option<Vec<Exchange>>) -> QueriedExchangeRateWithFailedExchanges {
```

Comment:
Shouldn't the parameter `exchanges` be called `failed_exchanges`?
Also, why do you use an `Option`? If the vector is empty, this means that there are no failed exchanges, right?